tests/vmi_lifecycle_test.go

```diff
@@ -220,17 +221,23 @@ var _ = Describe("[rfe_id:273][crit:high][arm64][vendor:cnv-qe@redhat.com][level
 				Should(ContainSubstring(`"subcomponent":"libvirt"`))
 		})
 
-		DescribeTable("log libvirtd debug logs should be", func(vmiLabels, vmiAnnotations map[string]string, expectDebugLogs bool) {
-			options := []libvmi.Option{libvmi.WithResourceMemory("32Mi")}
+		DescribeTable("log libvirtd debug logs should be", decorators.WgS390x, func(vmiLabels, vmiAnnotations map[string]string, expectDebugLogs bool) {
+			arch := util.TranslateBuildArch()
+			var memory string
+			if arch == "s390x" {
+				memory = "128Mi"
+			} else {
+				memory = "32Mi"
+			}
+			options := []libvmi.Option{libvmi.WithResourceMemory(memory)}
 			for k, v := range vmiLabels {
 				options = append(options, libvmi.WithLabel(k, v))
 			}
 
 			for k, v := range vmiAnnotations {
 				options = append(options, libvmi.WithAnnotation(k, v))
 			}
-			vmi := libvmi.New(options...)
-
+			vmi := libvmifact.NewGuestless(options...)
```

Comment:
Hi @lyarwood , Apoligies for unclear explanation from my side. Let me explain here clearly.
The libvmi.New function is not working for s390x. The VMI is not coming to running state. So, I am using Alpine for s390x.  So, I have moved the Alpine logic and libvmi.new(options) logic inside the NewGuestless function. 

```
func NewGuestless(opts ...libvmi.Option) *kvirtv1.VirtualMachineInstance {
	arch := util.TranslateBuildArch()
	// Use Alpine if the architecture is s390x.
	if checks.IsS390X(arch) {
		return NewAlpine(opts...)
	}
	return libvmi.New(opts...)
}
```
So, if we consider the above test. It requires 32Mi for the test to pass. If I directly call the libvmifact.NewGuestless without removing this part from NewGuestless function,
```
 opts = append(
		[]libvmi.Option{libvmi.WithResourceMemory(qemuMinimumMemory())},
```
 it will override the 32Mi which Iam passing in the function with QemuMinMemory i.e 1Mi and the test will fail.  So, I have removed this part of the code and explicitly passing the QemuMinMemory as a function argument.  So now the NewGuestless function will work as follows

```
NewGuestless(QemuMinMemory) --> it will work as the earlier  NewGuestless function 
NewGuestless(with our own memory and options). --> it will work as libvmi.New(options) if not s390x
NewGuestless(options) for s390x --> it will work as NewAlpine(options)
```

if i dont follow this way. I have to explicitly check for the s390x arch and call the NewAlpine fn.

```
if checks.IsS390x(arch) {
 vmi := libvmi.NewAlpine(options...)
}else{
  vmi := libvmi.New(options...)
}
```
I did this way because of this suggestion from one of the reviewer.
https://github.com/kubevirt/kubevirt/pull/12516#issuecomment-2309545558

Reply:
Thanks for taking the time to write that up but I'm still not convinced you need to change `NewGuestless` at all.

If `New` isn't working in this context for `s390x` then surely `New` or the test cases calling it that don't work on `s390x` should be changed.

That avoids the need to change every caller to `NewGuestless` and the very meaning of this helper (emphasis on it being _guestless_) as it's basically the same as `New` outside of `s390x` use with your changes.

Reply:
Hi @lyarwood ,  I have reverted the New Guestless function, and the files changes due to this to original.   I have also changed the test cases to reflect the changes.  Can you please review it again ?